src/data_structures/fmindex.rs

```diff
@@ -373,6 +373,7 @@ impl<DBWT: Borrow<BWT>, DLess: Borrow<Less>, DOcc: Borrow<Occ>> FMDIndex<DBWT, D
     }
 
     /// Backward extension of given interval with given character.
+    #[allow(clippy::many_single_char_names)]
```

Comment:
I mean, there's a reason this lint exists. On the one hand, it might be due to conventions in literature, on the other hand, clear names for variables is also helpful -- and in my opinion way more helpful, so perhaps leave this lint as a reminder to tackle that.

Reply:
I agree. Discussions should be made on the variable names. I've reverted these changes.